parcels/fieldset.py

```diff
@@ -41,10 +42,21 @@ class FieldSet:
     """
 
     def __init__(self, fields: list[Field | VectorField]):
-        # TODO Nick : Enforce fields to be list of Field or VectorField objects
+        for field in fields:
+            if not isinstance(field, (Field, VectorField)):
+                raise ValueError(f"Expected `field` to be a Field or VectorField object. Got {field}")
```

Comment:
Would we want this to say `Got {type(field)}`, consistent with what is used down in  line 128 ?

Reply:
I don't think so. Having `Got {field}` is nice since that would surface the type and the name of the field (which is important for the for loop)